test/nodetool/test_toppartitions.py

```diff
@@ -0,0 +1,182 @@
+                      Record('Victor Oladipo',    18,  0),
+                      Record('Andre Roberson',    2,   0),
+                      Record('Steven Adams',      2,   0)],
+        }
+        response = {
+            'read_cardinality': samplings['read_cardinality'],
+            'read': [record.to_json() for record in samplings['read']],
+            'write_cardinality': samplings['write_cardinality'],
+            'write': [record.to_json() for record in samplings['write']],
+        }
+    duration = 50000
+    capacity = 512
+    list_size = 20
+    table_filters = 'ks:cf'
+    options = {
+        "--duration": duration,
+        "-s": capacity,
+        "-k": list_size,
+        "--cf-filters": table_filters,
+    }
+    args = []
+    for name, value in options.items():
+        args.extend([name, str(value)])
+    params = {
+        'duration': str(duration),
+        'table_filters': 'ks:cf',
+        'capacity': str(capacity),
+    }
+    if request.config.getoption("nodetool") == "scylla":
+        # scylla sends list_size, while cassandra's nodetool does not.
+        params['list_size'] = str(list_size)
+    actual_output = nodetool("toppartitions", *args, expected_requests=[
+        expected_request("GET", "/storage_service/toppartitions",
+                         params=params,
+                         response=response),
+    ])
+
+    expected_output = ''
+    first = True
+    for operation in ('read', 'write'):
+        operation_upper = operation.upper()
+        cardinality = samplings[f'{operation}_cardinality']
+        expected_output += f'''\
+{operation_upper}S Sampler:
+  Cardinality: ~{cardinality} ({capacity} capacity)
+  Top {list_size} partitions:
+'''
+        if operation not in samplings:
+            expected_output += "\tNothing recorded during sampling period...\n"
+            continue
+
+        expected_output += Record.header()
+        for record in samplings[operation]:
+            expected_output += record.format()
+        if first:
+            expected_output += '\n'
+            first = False
+
+    assert normalize_samplings(actual_output) == normalize_samplings(expected_output)
+
+
+def test_toppartitions_invalid_capacity(nodetool, request):
+    options = {
+        "--duration": 50000,
+        "-s": 20,
+        "-k": 512,
+        "--cf-filters": 'ks:cf',
+    }
+    args = []
+    for name, value in options.items():
+        args.extend([name, str(value)])
+    with pytest.raises(subprocess.CalledProcessError) as e:
+        nodetool("toppartitions", *args)
+    assert e.value.returncode == 1
+    expected_output = "TopK count (-k) option must be smaller than the summary capacity (-s)"
+    if request.config.getoption("nodetool") == "scylla":
+        assert expected_output in e.value.stderr
+    else:
+        # cassandra nodetool's error message has a typo in it.
+        assert expected_output in e.value.output.replace('then', 'than')
```

Comment:
We have `util.nodetool_fails_with*()` to check error messages.

Reply:
great. i am using one of them in the latest revision.